lab5/app/src/main/java/com/app/lab5/Firebase_login.java

```diff
@@ -0,0 +1,135 @@
+                            et_email.requestFocus();
+                            return;
+
+                        }else if (et_pass.getText().toString().trim().isEmpty()) {
+                            Toast.makeText(this, "Password is required", Toast.LENGTH_LONG).show();
+                            et_pass.requestFocus();
+                            return;
+                        } else if (!et_pass.getText().toString().matches(passwordPattern)){
+                            et_pass.requestFocus();
+                            Toast.makeText(this, "Password must contain atleast 8 character long with one uppercase, one lowercase, one digit.", Toast.LENGTH_LONG).show();
+                            return;
+                        }else {
+                            Email = et_email.getText().toString();
+                            Password = et_pass.getText().toString();
+                            progressBar.setVisibility(View.VISIBLE);
+                            LoginFirebase();
+                        }break;
+                    case R.id.tv_signup2:
+                        startActivity(new Intent(Firebase_login.this, Firebase_signup.class));
+                        finish();
+                        break;
+                }
+
+            }
+
+            private void LoginFirebase(){
+                Login.signInWithEmailAndPassword(Email, Password)
+                        .addOnCompleteListener(Firebase_login.this, new OnCompleteListener<AuthResult>() {
+                            @Override
+                            public void onComplete(@NonNull Task<AuthResult> task) {
+                                // If sign in fails, display a message to the user. If sign in succeeds
+                                // the auth state listener will be notified and logic to handle the
+                                // signed in user can be handled in the listener.
+                                progressBar.setVisibility(View.GONE);
+                                if (!task.isSuccessful()) {
+                                    // there was an error
+                                    if (Password.length() < 6) {
+                                        Toast.makeText(Firebase_login.this, "Password must be atlest 6 character", Toast.LENGTH_LONG).show();
+                                    } else {
+                                        Toast.makeText(Firebase_login.this, "Authentication failed, check your email and password or sign up", Toast.LENGTH_LONG).show();
+                                    }
+                                } else {
+                                    Intent intent = new Intent(Firebase_login.this, mainmenu.class);
+                                    startActivity(intent);
+                                    finish();
+                                }
+                            }
+                        });
+
+
```

Comment:
Please remove all the unnecessary blank lines that decrease the readability of this file. The same for the other Firebase class